api/class/nestoria.js

```diff
@@ -0,0 +1,45 @@
+const fetch = require('node-fetch');
+const RawHome = require('../../db/model/rawHome');
+const connection = require('../../db/db');
+
+class NestoriaClient {
+  constructor() {
+    this.name = 'nestoria';
+    this.endpoint = 'https://api.nestoria.es/api?encoding=json&pretty=1&action=search_listings&country=es&listing_type=buy&place_name=barcelona';
+  }
+
+  async fetchData() {
+    const response = await fetch(this.endpoint, {
+      method: 'GET',
+    });
+    const responseJson = await response.json();
+    return responseJson.response.listings;
+  }
+
+  processData(rawHomesArray) {
+    const processedHomesArray = rawHomesArray.map((obj) => {
+      const processedObj = {};
```

Comment:
can you do this with destructuring syntax?

Reply:
there is definitely a better way of doing this! im not sure destructuring is the way though..
ill improve in the next version

Reply:
Well, yes… and no. It's not easy to refactor this. I think is your time to step into serializers. Serializers are classes/methods that format objects, usually between input and database. In this case you're receiving everything from an input (api) and storing it into the database.

In there you can destructure and structuring again but more lines, maybe more confusion… you can try it.

Reply:
nice nice! ive heard so much about serializers, so now ill look into it!